interop/deposit_contract_testnets/README.md

```diff
@@ -0,0 +1,68 @@
+
+https://github.com/eth2-clients/eth2-testnets
+
+Please create a sub-folder for your team, where each active testnet will be stored as a nested sub-folder with a descriptive name. For example: 
+
+```
+trinity/1000_validators_testnet/bootstrap_nodes.txt
+trinity/1000_validators_testnet/deposit_contract.txt
+trinity/1000_validators_testnet/config.yaml
+trinity/1000_validators_testnet/genesis.ssz
+```
+
+The `config.yaml` file is optional and it specifies the [Eth2 constants parameters](https://github.com/ethereum/eth2.0-specs/tree/dev/configs) used by the testnet. When no config file is present, it should be assumed that the testnet is using the `minimal` config. Users are expected to check each client's documentation for instructions regarding the usage of config preset files. Config presets are usually specified either at build-time or at run-time depending on the client.
+
+`bootstrap_nodes.txt` is a line-delimited text file with [multiaddr](https://github.com/multiformats/multiaddr) records for the bootstrap nodes of the testnet. For example:
+
+```
+/ip4/10.20.30.40/tcp/9100/p2p/16Uiu2HAmEAmp4FdpPzypKwTMmsbCdnUafDvXZCpFrUDbYJZNk7hX
+/ip4/10.20.30.50/tcp/9100/p2p/16Uiu2HAmV5SpcmrEymmaj1mymSHkbMpMVUXFRiXRxpHkfWdQ7bi3
+```
+
+To connect to the testnet, the user should either pass the entire file as a command-line parameter named `--bootstrap-file` or she should specify the individual entries with a repeatable command-line parameter named `--bootstrap-node`.
+
+`deposit_contract.txt` is a single-line text file with the address of the testnet's deposit contract on the Görli network. To connect to the testnet, the user should specify the address as a command-line parameter named `--deposit-contract`. If the used constants preset specifies a non-empty value for the `DEPOSIT_CONTRACT_ADDRESS` constant, it should be used as the default value for this parameter. The supplied address consists of 20 bytes encoded in hex form and prefixed with "0x".
+
+The `genesis.ssz` represents the genesis state snapshot. The user should specify the path to this file with a command-line parameter named `--state-snapshot`.
+
+Any additional information and instructions for interacting with the specific testnet can be provided in a README file in the same folder.
+
+### Making deposits
+
+To obtain the 32 GöETH required for making a deposit, please use the social faucet at https://faucet.goerli.mudit.blog/
+
+It's expected that the users will use a web-site or a command-line command provided by the client to make a deposit.
+
+### Creating testnets
+
+Please note that a testnet genesis file can feature a pre-populated validator set. The monitoring of Eth1 deposit events should start from the Eth1 block referenced in the `.eth1_data.eth1_block_hash` field of the genesis state. Thus, to create a testnet, you need to deploy a new deposit contract and then generate a genesis state that references a recent Görli block hash.
```

Comment:
even if we use a pre-populated validator set , we will also correspondingly need to send deposits with those validator's keys. Otherwise we would have differences in the deposit root in our local trie and in the contract. Also the deposit count we get from contract will be different from what we have in our current state. The monitoring of eth1 deposits should be from when the contract is deployed

Reply:
We can pre-populate if we agree on a set of validator keys, like we did at interop.

Doing so it would be valuable to perhaps "reserve" a set of keys per client, so those clients can reliably do validator tests.

Reply:
Yes, the initial keys are agreed upon in the sense that both clients load the same genesis state file.

The discrepancy with the data in the contract is a real problem, but our current implementation makes two compromises that seems reasonable for now:

1) It takes into account the number of validators in the genesis state when comparing the validator counts.

2) It monitors the contract only for deposit events. Only the locally computed root is considered valid. Since there is consensus on the latest Eth1 block, the set of deposit events in implicitly agreed upon as well.

Reply:
I like the idea of "reserving" mock keys for the different clients. I think something like this can be described in the [README file](https://github.com/eth2-clients/eth2-testnets/tree/master/nimbus/testnet0) of each network. For example, we can clarify the number of used mock keys and the suggested assignments (e.g. any key index for which `N % 10 == 6` is intended for LodeStar).

Reply:
> Please note that it's also possible to discover the number of used mock keys and their indices just by examining the public keys of the validators appearing in the genesis state.

For sure! Just a quick&dirtier solution might be to reserve them... /shrug

Reply:
I do like the idea of reserving keys for different clients to facilitate equal participation. However if we proceed having a mock genesis state with pre-populated validators we would need to also set a custom eth1Data for each genesis state that we create. Is there any reason we cant watch deposit events from contract deployment ? 
1) It would mirror more closely to how it would work in mainnet.
2) We wouldnt need to implement one off logic just for the purposes of running a multi-client testnet.
4) If we want to pre-populate with our desired validator set, that can be easily done by sending deposits with those keys included. For 64 deposits, it would take less than 2 minutes on Goerli for all of them to be mined.

Reply:
There was a little bit of additional reasoning behind this proposal that I didn't explicitly express. 

The ability to start from a trusted genesis file that includes a list of validators and a reference to an Eth1 block is something that each client will have to implement eventually in order to handle the situation where the user has been offline for a period greater than the weak subjectivity period. 

In this situation, you'll have to trust everything that you see in a supplied snapshot state. It's better to add additional sanity checks verifying that the Eth1 contract data is consistent with this trusted snapshot, but the compromise for now is to just not enforce these sanity checks.

For me, the most significant simplification from having pre-populated genesis states is not so much about the quick start-up times, but rather about reducing the required management of GoETH funds and their associated private keys.

Reply:
Afaik we do have ways to reduce the amount of goETH required and also we should be able to successfully recycle them for each testnet. For the purposes of interop starting from a common genesis state maybe best, we can look at watching deposit events from contract deployment maybe later.